src/k8s/pkg/k8sd/setup/kube_apiserver_test.go

```diff
@@ -28,5 +33,123 @@ func TestKubeAPIServer(t *testing.T) {
+		g.Expect(setup.KubeAPIServer(s, "10.0.0.0/24", "https://auth-webhook.url", false, "k8s-dqlite", "Node,RBAC")).To(BeNil())
+
+		// Ensure the kube-apiserver arguments file has the expected arguments and values
+		tests := []struct {
+			key         string
+			expectedVal string
+		}{
+			{key: "--allow-privileged", expectedVal: "true"},
+			{key: "--authentication-token-webhook-config-file", expectedVal: path.Join(s.Mock.ServiceExtraConfigDir, "auth-token-webhook.conf")},
+			{key: "--authorization-mode", expectedVal: "Node,RBAC"},
+			{key: "--client-ca-file", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "ca.crt")},
+			{key: "--enable-admission-plugins", expectedVal: "NodeRestriction"},
+			{key: "--kubelet-certificate-authority", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "ca.crt")},
+			{key: "--kubelet-client-certificate", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "apiserver-kubelet-client.crt")},
+			{key: "--kubelet-client-key", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "apiserver-kubelet-client.key")},
+			{key: "--kubelet-preferred-address-types", expectedVal: "InternalIP,Hostname,InternalDNS,ExternalDNS,ExternalIP"},
+			{key: "--secure-port", expectedVal: "6443"},
+			{key: "--service-account-issuer", expectedVal: "https://kubernetes.default.svc"},
+			{key: "--service-account-key-file", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "serviceaccount.key")},
+			{key: "--service-account-signing-key-file", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "serviceaccount.key")},
+			{key: "--service-cluster-ip-range", expectedVal: "10.0.0.0/24"},
+			{key: "--tls-cert-file", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "apiserver.crt")},
+			{key: "--tls-cipher-suites", expectedVal: apiserverTLSCipherSuites},
+			{key: "--tls-private-key-file", expectedVal: path.Join(s.Mock.KubernetesPKIDir, "apiserver.key")},
+			{key: "--etcd-servers", expectedVal: fmt.Sprintf("unix://%s", path.Join(s.Mock.K8sDqliteStateDir, "k8s-dqlite.sock"))},
+		}
+		for _, tc := range tests {
+			t.Run(tc.key, func(t *testing.T) {
+				val, err := snaputil.GetServiceArgument(s, "kube-apiserver", tc.key)
+				g.Expect(err).To(BeNil())
+				g.Expect(val).To(Equal(tc.expectedVal))
+			})
+		}
+
+		// Ensure the kube-apiserver arguments file has exactly the expected number of arguments
+		args, err := utils.ParseArgumentFile(path.Join(s.Mock.ServiceArgumentsDir, "kube-apiserver"))
+		g.Expect(err).To(BeNil())
+		g.Expect(len(args)).To(Equal(len(tests)))
+	})
+
+	t.Run("Handle unsupported datastore", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Create a mock snap
+		s, _ := testApiServerFixture(t)
+
+		// Attempt to configure kube-apiserver with an unsupported datastore
+		err := setup.KubeAPIServer(s, "10.0.0.0/24", "https://auth-webhook.url", false, "unsupported-datastore", "Node,RBAC")
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(err.Error()).To(ContainSubstring("unsupported datastore"))
```

Comment:
TIL: https://github.com/onsi/gomega/blob/master/matchers.go#L114

`g.Expect(err).To(MatchError(ContainsSubstring("unsupported datastore"))`